### ISSUE TITLE
no-commonjs: Allow assigning to a variable called exports

### ISSUE THREAD

**Issue by SimenB** [benmosher/eslint-plugin-import#940]
`exports` should only be disallowed assignment if it's the "global" `exports` of CJS, not any old variable named `exports`.

```js
function someFunc() {
  const exports = someComputation();

  expect(exports.someProp).toEqual({ a: 'value' });
}
```

![image](https://user-images.githubusercontent.com/1404810/31319390-d6c52c76-ac62-11e7-9c02-22f6181ed641.png)

**Comment by adrianhelvik** [comment]
See pull request #978 for a fix.
